src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -527,14 +527,10 @@ public static CSharpCodeWritingScope BuildClassDeclaration(
 
             if (hasBaseType)
             {
-                if (baseType.Source is { } source)
-                {
-                    WriteWithPragma(writer, baseType.Content, context, source);
-                }
-                else
-                {
-                    writer.Write(baseType.Content);
-                }
+                WriteOptionalToken(baseType.BaseType);
```

Comment:
I could understand the base type not being generic, but when will the base type be null entirely?

Reply:
It shouldn't be, I was just being lazy using the same method. I split it out into `WriteToken` and `WriteOptionalToken` and updated the call sites (applies for interfaces comment too).